### DISCUSSION TITLE
Offline Robot window

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#4443] [Q&A]
Hello,

AFAIK, when using the robot window, since it's running in the browser, it need a script (https://cyberbotics.com/wwi/R2022b/RobotWindow.js)  and an image (https://cyberbotics.com/assets/images/webots.png) to make the whole thing work.
However, instead of being locally available, they are download from cyberbotics.com
Could it be possible to bundle them with Webots, so that we don't need internet connection (and ping the server for files) to run the Robot Window ?

Thanks !

**Comment by omichel** [comment]
HTML robot windows should be easily deployable on the web. Thus, they should not refer to local Webots files, but to URLs. If you want, you can easily get rid of this by making local copies of the URLs and referring to them in your custom robot window.

**Reply by ShuffleWire** [reply]
But the robot windows already relies on many local script, what make this script different from the others ? :/ it's 9 scripts that are already local : http://localhost:1234/robot_windows/generic/{generic, window_manager, device_widget, radar_widget, timeplot_widget, timeplot_line, canvas, plot_widget}.js
In the end it's not a big deal, I just want to ensure it haven't been forgotten somewhere during development !

**Reply by BenjaminDeleze** [reply]
The difference is that the 9 scripts are specific to this robot window in particular: the generic robot window. Whereas RobotWindow.js is used by all robot windows.

**Reply by ShuffleWire** [reply]
Ok, got it !
